netlayers/tcp.py

```diff
@@ -23,15 +23,31 @@
 
 class TCPNetlayer(Netlayer):
 
-    def __init__(self, listen_address="0.0.0.0", listen_port=22045, public_address="0.0.0.0", public_port=22045):
-        self.address, self.port = listen_address, listen_port
-        self.location = OCapNMachine(syrup.Symbol("tcp"), f"{public_address}:{public_port}", False)
-
-        self._connections = []
-
+    def __init__(self, 
+        listen_address="0.0.0.0", 
+        listen_port=22045, 
+        public_address="0.0.0.0", 
+        public_port=22045, 
+        autoport=True
+    ):
         self.server_sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        self.server_sock.bind((listen_address, listen_port))
 
+        # Trivial search for a next open port
+        if autoport:
+            while True:
```

Comment:
add <65536 condition

Reply:
can't you just bind to port 0 and get one from the system?

Reply:
yes we can (=

```python
import socket
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.bind(('0.0.0.0', 0))
s.getsockname()[1]
50189
```